gnuradio3_8: remove

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
gnuradio: remove unused swig argument

doronbehar committed:
gnuradioMinimal: move to pkgs/by-name

doronbehar committed:
qradiolink: switch to GR 3.10

doronbehar committed:
inspectrum: switch to GR 3.10

doronbehar committed:
gnuradio3_8: remove

doronbehar committed:
swig3: remove

doronbehar committed:
gnuradio: use new apple-sdk pattern

emilazy commented:
The `qradiolink` failure seems like it might relate to `spdlog` being in `-I` rather than `-isystem`, making the nested include fail.

doronbehar committed:
gnuradioPackages.osmosdr: don't use with lib; in meta

doronbehar committed:
gnuradioPackages.osmosdr: simplify expression for a single GR version

Remove unused swig3 dependency - was really needed only for GR 3.8 that
is now removed.

doronbehar committed:
gnuradioPackages.ais: remove

Cannot be used anymore by any modern GR version.

doronbehar committed:
gnuradioPackages.grnet: remove

Cannot be used anymore by any modern GR version.

doronbehar committed:
gnuradioPackages.limesdr: remove

Cannot be used anymore by any modern GR version.

doronbehar committed:
gnuradioPackages.rds: remove

Cannot be used anymore by any modern GR version.

doronbehar committed:
gnuradioPackages.nacl: remove

Cannot be used anymore by any modern GR version.

doronbehar committed:
gnuradioPackages.gsm: remove

Cannot be used anymore by any modern GR version.

markuskowa commented:
I did not have time to take a closer look, but there seems to be a branch of qradiolink that is compatible with gr-3.10:
https://github.com/qradiolink/qradiolink/tree/gr_3.10

doronbehar commented:
Thanks for the notification @emilazy . I won't mind wait until that PR reaches `master` to check again. In the meantime I resolved the merge conflicts.

doronbehar commented:
OK now that `staging-next` is pretty baked, I included https://github.com/NixOS/nixpkgs/pull/365839 and I noticed I get different GR headers related errors. Then I switched qradiolink to the `gr_3.10` branch, and I only had to add `log4cpp` to the `buildInputs` and the build succeeded (upstream is probably still relying on the old log library although GR is using `spdlog` nowadays).

@markuskowa if you want to test `qradiolink` before `staging-next` reaches `master`, you should be able to simply `rebase` this PR to current `staging-next` (but don't forget to also include https://github.com/NixOS/nixpkgs/pull/365839 ). I think this PR should be ready once `staging-next` reaches `master.

doronbehar commented:
Tested that qradiolink builds, launches and GUI seems functional.

doronbehar commented:
staging-next was merged:

- #361878

doronbehar commented:
Converted back to draft due to:

- https://github.com/xbmc/xbmc/issues/26164

Which I hope kodi developers will solve. cc @NixOS/kodi .

doronbehar committed:
kodi: add patch for swig 4.3.0

doronbehar commented:
Fixed kodi issue.

paparodeo commented on pkgs/applications/radio/gnuradio/default.nix:
```diff
@@ -21,7 +20,7 @@
 , libiio
 , libad9361
-, CoreAudio
+, apple-sdk
```
```suggestion
```

paparodeo commented on pkgs/applications/radio/gnuradio/default.nix:
```diff
@@ -173,7 +172,7 @@ let
       runtime = []
         ++ lib.optionals stdenv.hostPlatform.isLinux [ alsa-lib libjack2 ]
-        ++ lib.optionals stdenv.hostPlatform.isDarwin [ CoreAudio ]
+        ++ lib.optionals stdenv.hostPlatform.isDarwin [ apple-sdk ]
```
```suggestion
```

paparodeo commented on pkgs/development/gnuradio-modules/osmosdr/default.nix:
```diff
@@ -1,6 +1,6 @@
 { lib
 , stdenv
-, darwin
+, apple-sdk
```
```suggestion
```

paparodeo commented on pkgs/development/gnuradio-modules/osmosdr/default.nix:
```diff
@@ -72,8 +61,7 @@ in mkDerivation {
   ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    darwin.apple_sdk.frameworks.IOKit
-    darwin.apple_sdk.frameworks.Security
+    apple-sdk
```
```suggestion
```

paparodeo reviewed (commented):
LGTM -- don't need to add `apple-sdk` as that comes with the stdenv. if you need an sdk other than the default you'd do something like `apple-sdk_15` which would bump the sdk to version 15 rather than 11, the current default for both platforms.

doronbehar commented:
Thanks for the review @paparodeo :). Done.

paparodeo reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356976`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>gnuradio</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>trunk-recorder</li>
  </ul>
</details>

doronbehar commented:
Friendly ping @markuskowa

markuskowa reviewed (approved):
Thanks for fixing everything.

emilazy commented:
Thanks for tackling this!
